### PR TITLE
feat(console): user personal access tokens

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement user personal access tokens (PAT) management in Console.

## Testing

截屏2024-08-14 14 41 38

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/index.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  );
+  const api = useApi();
+
+  const tableColumns: Array<Column<Token>> = useMemo(
+    () => [
+      {
+        title: t('general.name'),
+        dataIndex: 'name',
+        colSpan: 3,
+        render: ({ name }) => <span>{name}</span>,
```
Comment (xiaoyijun) [inline packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/index.tsx:37]:
```suggestion
        render: ({ name }) => <Breakable>{name}</<Breakable>,
```

File: packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/CreateTokenModal.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  const [expirationDate, setExpirationDate] = useState<Date>();
+
+  useEffect(() => {
+    const setDate = () => {
+      if (expirationDays === neverExpires) {
+        setExpirationDate(undefined);
+      } else {
+        setExpirationDate(addDays(new Date(), Number(expirationDays)));
+      }
+    };
```
Comment (xiaoyijun) [inline packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/CreateTokenModal.tsx:58]: Update the this state in the `onChange` callback from the `Select` component?

Comment (wangsijie) [reply packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/CreateTokenModal.tsx:58]: There is an interval to clear, so `useEffect` would be easier.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**